src/concrete_solve.jl

```diff
@@ -334,6 +334,211 @@ function DiffEqBase._concrete_solve_adjoint(prob, alg,
     out, adjoint_sensitivity_backpass
 end
 
+function DiffEqBase._concrete_solve_adjoint(prob, alg,
+                                            sensealg::AbstractAdjointSensitivityAlgorithm,
+                                            u0::StaticArrays.SVector, p, originator::SciMLBase.ADOriginator,
+                                            args...; save_start = true, save_end = true,
+                                            saveat = eltype(prob.tspan)[],
+                                            save_idxs = nothing,
+                                            kwargs...)
+    if !(typeof(p) <: Union{Nothing, SciMLBase.NullParameters, AbstractArray}) ||
+        (p isa AbstractArray && !Base.isconcretetype(eltype(p)))
+        throw(AdjointSensitivityParameterCompatibilityError())
+    end
+
```

Comment:
maybe add an `@assert sensealg isa QuadratureAdjoint` if we only implement it for this case and throw an error otherwise.

Reply:
throw a custom error like above but with a custom error message describing that `!ArrayInterface.ismutable(u0)` types require QuadratureAdjoint. Then we should make an issue to extend it afterwards, but put a low priority on it because of the issue with parameter sizing.